src/object/objecttype.c

```diff
@@ -341,6 +341,9 @@ bool_t CONST sameRegionAs(cap_t cap_a, cap_t cap_b)
 
     case cap_irq_control_cap:
         if (cap_get_capType(cap_b) == cap_irq_control_cap ||
+#if CONFIG_MAX_NUM_NODES == 1
+            cap_get_capType(cap_b) == cap_sgi_signal_cap ||
+#endif
```

Comment:
Do some of these preprocessor guards in arch-generic code need to be restricted to `CONFIG_ARCH_ARM`? Or is the idea that every arch would eventually get an SGI signal cap?